workers/unity/Assets/Fps/Scripts/GameLogic/WorldTiles/TileTypeVolume.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Fps;
+using UnityEngine;
+
+[RequireComponent(typeof(BoxCollider))]
+public class TileTypeVolume : MonoBehaviour
+{
+    public TileTypeCollection TypeCollection;
+    public bool AutoRename;
+    private BoxCollider boxCollider;
+
+    private void Awake()
+    {
+        boxCollider = GetComponent<BoxCollider>();
+    }
+
+    private void OnValidate()
+    {
+        if (!AutoRename)
+        {
+            return;
+        }
+
+        gameObject.name = TypeCollection == null ? "Empty TileTypeCollection" : $"{TypeCollection.name} TypeVolume";
+    }
+
+    private void OnDrawGizmos()
+    {
+        if (TypeCollection == null)
+        {
+            return;
+        }
+
+        if (boxCollider == null)
```

Comment:
Can `OnDrawGizmos` happen before `Awake` ?

Reply:
Yes, in editor mode `OnDrawGizmos` is called but `Awake` isn't.

Reply:
(Unless `[ExecuteInEditMode]` attribute is added to the class, but that doesn't make sense here)

Reply:
Cool - just wanted to double check 😄